src/util/get-update-command.ts

```diff
@@ -24,10 +24,91 @@ async function isYarn(): Promise<boolean> {
   return !('_id' in pkg);
 }
 
+async function canRead(file: string) {
+  return new Promise((res) => {
+    fs.access(file, fs.constants.F_OK | fs.constants.R_OK, (err) => {
+      if (err) {
+        res(false);
+        return;
+      }
+
+      res(true);
+    });
+  });
+}
+
+async function getConfigPrefix() {
+  const paths = [
+    process.env.npm_config_userconfig || process.env.NPM_CONFIG_USERCONFIG,
+    join(process.env.HOME || '/', '.npmrc'),
+    process.env.npm_config_globalconfig || process.env.NPM_CONFIG_GLOBALCONFIG
+  ].filter(Boolean);
+
+  for (const configPath of paths) {
+    if (!configPath) {
+      continue;
+    }
+
+    if (await canRead(configPath) === false) {
```

Comment:
I'd prefer to just do the `readFile()` operation, and then handle the errors appropriately in a try/catch.